File: problem/4/python/solve_4.py
#!/usr/bin/env python
# http://projecteuler.net/problem=4

import sys

# create palindromes
max_palindrome = 998001
min_palindrome =  10000
temp_palindrome = max_palindrome
palindrome_list = []

def is_palindrome(p):
        temp = list(str(p))
        temp_rev = temp[:]
        temp_rev.reverse()
        if temp == temp_rev:
                return True
        else:
                return False

while temp_palindrome >= min_palindrome:
        if is_palindrome(temp_palindrome):
                palindrome_list.append(temp_palindrome)
        temp_palindrome -= 1

for x in palindrome_list:
        for y in range(100,1000):
                if x % y == 0:
                        if x/y in range(100, 1000):
                                print "palindrome: " + str(x)
                                print "factors: " + str(y) + " " + str(x/y)
                                sys.exit(0)

File: problem/4/perl/solve_4.pl
#!/usr/bin/env perl6
# http://projecteuler.net/problem=4

use v6;

my $max_palindrome = 997;
my $min_palindrome =  100;

loop (my $temp_palindrome = $max_palindrome; $temp_palindrome >= $min_palindrome; $temp_palindrome--) {
        my $x = $temp_palindrome ~ $temp_palindrome.split('').reverse().join();
        for 100 .. 999 -> $y {
                my $z = $x/$y;
                if ($z.Int == $z) and ($z ~~ 100 .. 999)  {
                        say "Palindrome: " ~ $x;
                        say "Factors:    " ~ $y ~ " " ~ $z;
                        exit 0;
                }
        }
}
